### PR TITLE
Svgo v2

### PR DESCRIPTION
**Pull request by Aissaoui-Ahmed**
Ref: #307

NodeJS [releases](https://nodejs.org/en/about/releases/) status and END-OF-LIFE

error svgo@2.8.0: The engine "node" is incompatible with this module. Expected version ">=10.13.0". Got "8.17.0"
https://github.com/svg/svgo/blob/b37d90e12a87312bba87a6c52780884e6e595e23/package.json#L52-L54

Travis build
https://app.travis-ci.com/github/boopathi/react-svg-loader/jobs/550900558

### CODE REVIEW

File: packages/react-svg-core/src/index.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
-import Svgo from "svgo";
 import { transformSync as babelTransform } from "@babel/core";
 import plugin from "babel-plugin-react-svg";
-
+import Svgo from "svgo/lib/svgo/jsAPI";
```
Comment (TrySound) [inline packages/react-svg-core/src/index.ts]: It's not right. `svgo` package provides `optimize` named export which no longer output promise. Check this https://github.com/svg/svgo/releases/tag/v2.0.0

File: packages/react-svg-core/src/index.ts [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
 import { transformSync as babelTransform } from "@babel/core";
 import plugin from "babel-plugin-react-svg";
-
+import { optimize } from "svgo";
 import { validateAndFix } from "./svgo";
 
+
 // SVGO Optimize
-export function optimize(opts: any = {}): (content: string) => Promise<string> {
+export function optimize(opts: any = {}) {
```
Comment (TrySound) [inline packages/react-svg-core/src/index.ts:7]: Conflicting variable name

File: packages/react-svg-core/src/index.ts [hunk 3]
```diff
@@ -1,15 +1,13 @@
...
   opts = validateAndFix(opts);
-  const svgo = new Svgo(opts);
-
-  return (content: string) => svgo.optimize(content).then(data => data.data);
+  return (content: string) => optimize(content);
```
Comment (TrySound) [inline packages/react-svg-core/src/index.ts]: options are not passed

File: packages/react-svg-core/src/index.ts [hunk 4]
```diff
@@ -1,15 +1,13 @@
...
 import plugin from "babel-plugin-react-svg";
-
+import { optimize } from "svgo";
 import { validateAndFix } from "./svgo";
 
+
 // SVGO Optimize
-export function optimize(opts: any = {}): (content: string) => Promise<string> {
+export function optimize(opts: any = {}) {
   opts = validateAndFix(opts);
```
Comment (TrySound) [inline packages/react-svg-core/src/index.ts]: Config format is different now. So validateAndFix might need to be changed as well.

**Comment by a-viking** [comment]
@boopathi can you review this, please? We use your loader and it fails vulnerability check every build.

**Pull request closed without merging**